### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by IdanBanani** [tim-harding/autograder#1]
Release option on Ubuntu 20.04 (glibc 2.31):
```
./autograder
./autograder: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./autograder
```

Cargo option:
```
cargo install --git https://github.com/tim-harding/autograder
    Updating git repository `https://github.com/tim-harding/autograder`
  Installing autograder v0.3.0 (https://github.com/tim-harding/autograder#ac3cf9f5)
    Updating crates.io index
  ...............
 Compiling clap v3.0.0-beta.5
   Compiling autograder v0.3.0 (/home/idan/.cargo/git/checkouts/autograder-28301cd75abab5d1/ac3cf9f)
error[E0432]: unresolved import `clap::Clap`
 --> src/main.rs:1:5
  |
1 | use clap::Clap;
  |     ^^^^^^^^^^ no `Clap` in the root

error: cannot determine resolution for the derive macro `Clap`
  --> src/main.rs:15:10
   |
15 | #[derive(Clap, Debug, Clone, Hash, PartialEq, Eq)]
   |          ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `clap` in this scope
  --> src/main.rs:16:3
   |
16 | #[clap(version = "0.3.0", author = "Tim Harding <Tim@TimHarding.co>")]
   |   ^^^^

error: cannot find attribute `clap` in this scope
  --> src/main.rs:19:7
   |
19 |     #[clap(short, long, default_value = "./.github/classroom/autograding.json")]
   |       ^^^^

error: cannot find attribute `clap` in this scope
  --> src/main.rs:22:7
   |
22 |     #[clap(short, long)]
   |       ^^^^

error[E0599]: no function or associated item named `parse` found for struct `Options` in the current scope
   --> src/main.rs:136:37
    |
17  | struct Options {
    | -------------- function or associated item `parse` not found for this
...
136 |     let options: Options = Options::parse();
    |                                     ^^^^^ function or associated item not found in `Options`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `parse`, perhaps you need to implement it:
            candidate #1: `Parser`

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: failed to compile `autograder v0.3.0 (https://github.com/tim-harding/autograder#ac3cf9f5)`, intermediate artifacts can be found at `/tmp/cargo-install8qiYDc`

Caused by:
  could not compile `autograder` due to 6 previous errors
```

After installing clap:
```
`error[E0432]: unresolved import `clap::Clap`
 --> src/main.rs:1:5
  |
1 | use clap::Clap;
  |     ^^^^^^^^^^ no `Clap` in the root

error: cannot determine resolution for the derive macro `Clap`
  --> src/main.rs:15:10
   |
15 | #[derive(Clap, Debug, Clone, Hash, PartialEq, Eq)]
   |          ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `clap` in this scope
  --> src/main.rs:16:3
   |
16 | #[clap(version = "0.3.0", author = "Tim Harding <Tim@TimHarding.co>")]
   |   ^^^^

error: cannot find attribute `clap` in this scope
  --> src/main.rs:19:7
   |
19 |     #[clap(short, long, default_value = "./.github/classroom/autograding.json")]
   |       ^^^^

error: cannot find attribute `clap` in this scope
  --> src/main.rs:22:7
   |
22 |     #[clap(short, long)]
   |       ^^^^

error[E0599]: no function or associated item named `parse` found for struct `Options` in the current scope
   --> src/main.rs:136:37
    |
17  | struct Options {
    | -------------- function or associated item `parse` not found for this
...
136 |     let options: Options = Options::parse();
    |                                     ^^^^^ function or associated item not found in `Options`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `parse`, perhaps you need to implement it:
            candidate #1: `Parser`

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: failed to compile `autograder v0.3.0 (https://github.com/tim-harding/autograder#ac3cf9f5)`, intermediate artifacts can be found at `/tmp/cargo-installyQIsUa`
```

Caused by:
  could not compile `autograder` due to 6 previous errors
`
By the way: does Github releases a tool/script that runs tests inside the autogradings.json? (free for everybody?)

**Comment by tim-harding** [comment]
Thanks for the report. This is what happens when you use Git releases rather than stable ones from crates.io, I guess. Or when you don't link to a specific commit, at least.  Clap changed their API on main. I'll get around to fixing it sometime soon. Per your question, I don't think that GitHub publishes their autograder. They went closed source with the project a while ago (ironically!). That's why I created this originally.

**Comment by tim-harding** [comment]
Fixed with e049abb39eb8d297bd754e8c48bdb2e404662e04
